Presentation/UINavigationController+Presenting.swift

```diff
@@ -156,21 +162,31 @@ private extension UINavigationController {
             animated = animated && viewControllers.count > 0
         }
 
+
         if let coordinator = transitionCoordinator, !animated {
             // If we update the vcs while the nc (self) is being presented, the nc gets lost and controls in the the presented vcs can't become first responders.
             // Moving presentation inside transition animate fixes issue.
             let willAnimate = coordinator.animate(alongsideTransition: { context in
                 let wasEnabled = UIView.areAnimationsEnabled
                 UIView.setAnimationsEnabled(false)
                 self.setViewControllers(vcs, animated: false)
+                if let navBarHidden = navBarHidden {
+                    self.setNavigationBarHidden(navBarHidden, animated: false)
+                }
                 UIView.setAnimationsEnabled(wasEnabled)
             })
 
             if !willAnimate { // if the animation block wont't be called, fallback to normal setting up of vcx
                 setViewControllers(vcs, animated: animated)
+                if let navBarHidden = navBarHidden {
+                    self.setNavigationBarHidden(navBarHidden, animated: animated)
+                }
             }
         } else {
             setViewControllers(vcs, animated: animated)
+            if let navBarHidden = navBarHidden {
+                self.setNavigationBarHidden(navBarHidden, animated: animated)
+            }
         }
```

Comment:
Not sure I follow all updates to `setNavigationBarHidden` above. Should it not just be enough to update based on the latest topViewController (vcs.last)?